statistics/handle/cache/internal/mapcache/bench_test.go

```diff
@@ -0,0 +1,76 @@
+// Copyright 2023 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package mapcache
+
+import (
+	"sync"
+	"testing"
+
+	"github.com/pingcap/tidb/statistics/handle/cache/internal/testutil"
+	"github.com/pingcap/tidb/util/benchdaily"
+)
+
+func BenchmarkMapCachePut(b *testing.B) {
+	var (
+		wg sync.WaitGroup
+		c  = NewMapCache()
+	)
+
+	b.ResetTimer()
+	for i := 0; i < b.N; i++ {
+		wg.Add(1)
+		go func(i int) {
+			defer wg.Done()
+			t1 := testutil.NewMockStatisticsTable(1, 1, true, false, false)
+			c.Put(int64(i), t1, true)
+		}(i)
+	}
+	wg.Wait()
+	b.StopTimer()
+}
+
+func BenchmarkMapCachePutGet(b *testing.B) {
+	var (
+		wg sync.WaitGroup
+		c  = NewMapCache()
+	)
+	b.ResetTimer()
+
+	for i := 0; i < b.N; i++ {
+		wg.Add(1)
+		go func(i int) {
+			defer wg.Done()
+			t1 := testutil.NewMockStatisticsTable(1, 1, true, false, false)
```

Comment:
unfortunately, When using the ```CopyAndUpdate```, it will use the table id of structure.
so we make this structure difference. So it impacts each benchmark, which means it has no impact.

Reply:
we need to insert the map and not update the map. so we need to make this structure difference.